File: FangZhiHu/app/src/main/java/network/api/ApiManagerHelp.java
package network.api;


import com.tongqinyuan.fangzhihu.api.LoanService;

import network.RetrofitHelp;

/**
 * Created by yuantongqin on 2017/8/7.
 */

public class ApiManagerHelp {


    private final static Object operation = new Object();

    private static RetrofitHelp retrofitHelp;
    private static RetrofitHelp retrofitDownload;


    //获取下载api
    public static <T> T getDownloadApiClass(Class<T> tClass) {
        synchronized (operation) {
            if (retrofitDownload == null) {
                retrofitDownload = new RetrofitHelp(RetrofitHelp.DOWNLOAD);
            }
            return retrofitDownload.getRetrofit().create(tClass);
        }
    }

    //获取api
    public static LoanService getApiClass() {
        synchronized (operation) {
            if (retrofitHelp == null) {
                retrofitHelp = new RetrofitHelp(RetrofitHelp.EMPTY);
            }
            return retrofitHelp.getRetrofit().create(LoanService.class);
        }
    }

}

File: FangZhiHu/CommonLibKotlin/src/main/java/network/ApiManagerHelp.kt
package network

import network.api.ApiService

/**
 * Created by tongqinyuan on 2018/3/6.
 */
object ApiManagerHelp {

    private val operation = Any()

    private var mRetrofitHelp: RetrofitHelp? = null
    private var mRetrofitDownloan: RetrofitHelp? = null


    fun <T> getDownloadApiClass(tClass: Class<T>): T {
        synchronized(operation) {
            if (mRetrofitDownloan == null) {
                mRetrofitDownloan = RetrofitHelp(RetrofitHelp.DOWNLOAD)
            }

            return mRetrofitDownloan!!.mRetrofit.create(tClass)
        }

    }

    fun <T> getLoanService(tClass: Class<T>): T {
        synchronized(operation) {
            if (mRetrofitHelp == null) {
                mRetrofitHelp = RetrofitHelp(RetrofitHelp.EMPTY)
            }
            return mRetrofitHelp!!.mRetrofit.create(tClass)
        }
    }



}